packages/grid/data-grid/src/useDataGridProps.ts

```diff
@@ -10,6 +10,16 @@ export const useDataGridProps = (inProps: DataGridProps): GridComponentProps =>
 
   const themedProps = useThemeProps({ props: inProps, name: 'MuiDataGrid' });
 
+  const classes = React.useMemo(() => {
+    if (!inProps.classes) {
+      return {};
+    }
+    return Object.entries(inProps.classes).reduce((acc, [key, value]) => {
```

Comment:
Can we comment why we need to replace `__` by `--` ?

Reply:
It's ugly to use two dashes in an interface member. See https://github.com/mui-org/material-ui-x/issues/420#issuecomment-896359219. Now the replace is being done in `generateUtilityClass` and this pre-process doesn't exist anymore.